projects/packages/connection/src/class-tokens.php

```diff
@@ -592,4 +607,71 @@ public function get_user_tokens() {
 	public function update_user_tokens( $tokens ) {
 		return Jetpack_Options::update_option( 'user_tokens', $tokens );
 	}
+
+	/**
+	 * Lock the tokens to the current site URL.
+	 *
+	 * @param int $timespan How long the tokens should be locked, in seconds.
+	 *
+	 * @return bool
+	 */
+	public function set_site_lock( $timespan = HOUR_IN_SECONDS ) {
```

Comment:
My reasoning was that the whole class is dedicated to handling tokens, so `( new Tokens() )->set_site_lock()` seemed to be a reasonable name.
But yeah, it might make someone think that it's locking the whole site, not just locking the tokens to this particular site.
I'll rename the method.

Reply:
Completed https://github.com/Automattic/jetpack/pull/23597/commits/4dad2653c6cfc64be5cdd8b4cfa526f68c4a5834